pandas/tests/io/formats/style/test_format.py

```diff
@@ -192,13 +192,22 @@ def test_format_escape_html(escape, exp):
     assert styler._translate(True, True)["head"][0][1]["display_value"] == f"&{exp}&"
 
 
-def test_format_escape_latex_math():
-    chars = r"$\frac{1}{2} \$ x^2$ ~%#^"
-    df = DataFrame([[chars]])
+@pytest.mark.parametrize(
+    "chars, expected",
+    [
+        (r"$\frac{1}{2} \$ x^2$ ", r"$\frac{1}{2} \$ x^2$ "),
+        (r"\(\frac{1}{2} \$ x^2\) ", r"\(\frac{1}{2} \$ x^2\) "),
+        (r"\)", r"\) "),
+    ],
+)
```

Comment:
Am I correct in thinking that this will only escape latex math escape dollar signs or brackets, per DataFrame cell. So one cell could have `$ \frac{1}{2} $` and another cell could have `\( \frac{1}{2} \)` and each will be escaped properly, independently, but if the combination was applied in the same cell then only dollar signs would apply.
I.e. I think that, within your test, for, 
```
chars = "$ \frac{1}{2} $ \( \frac{1}{2} \)"
expected = "$ \frac{1}{2} $ \\( \\frac\{1\}\{2\} \\)"
```
It might be possible to get a test result that has the following,
```
expected = "$ \frac{1}{2} $ \( \frac{1}{2} \)"
```
but that will involve more complicated code, and it may be sufficient to go with what you have and just to document this. Whichever you choose please add this test with a comment on what its testing.

Also I think it would be better to add these tests, in (chars, expected) format:
```python
conversion = r"\&\%\#\_\{\}\textasciitilde \textasciicircum \textbackslash "
(r"$ &%#_{}~^\ $ &%#_{}~^\ $", f"$ &%#_{}~^\ $ {conversion} $")
```
Notice that this test has three $'s so i assume it will detect the first block first. Should repeat with:
```python
(r"\( &%#_{}~^\ \) &%#_{}~^\ \)", f"\( &%#_{}~^\ \) {conversion} \)")
```

Reply:
Thank you for the comments.
It’s correct, latex math escape works for each DataFrame cell separately. If we have a combination of dollar signs and brackets, the sign which occurs first would apply. At least that’s what I wanted to implement.

My result for the chars isn’t equal to the expected
```
chars = "$ \frac{1}{2} $ \( \frac{1}{2} \)"
result = $ \frac{1}{2} $ \(  \textbackslash frac\{1\}\{2\} \)
expected = "$ \frac{1}{2} $ \\( \\frac\{1\}\{2\} \\)"
```

It means there is a mistake and I should correct  `_escape_latex_math` and maybe `_escape_latex`. Then I will add this test.

Reply:
no you are correct. My brief message was wrong and poorly researched. `textbackslash` is correct.

I am surprised however that your result has not escaped the `$`s and its math contents but has also not escaped the parentheses but has escaped the contents, i.e. I would have expected:

`$ \frac{1}{2} $ \textbackslash (  \textbackslash frac\{1\}\{2\} \textbackslash )`

Reply:
I completely agree, the expected result should be 

`$ \frac{1}{2} $ \textbackslash (  \textbackslash frac\{1\}\{2\} \textbackslash )
`

My function didn't work right and I fixed it. I added the examples that you suggested to my test in (chars, expected) format.

I thought it would be good to add a couple of examples to docs. What do you think?